internal/controller/management_controller.go

```diff
@@ -772,6 +775,34 @@ func updateComponentsStatus(
 	}
 }
 
+// setReadyCondition updates the Management resource's "Ready" condition based on whether
+// all components are healthy.
+func setReadyCondition(management *kcm.Management) {
+	var failing []string
+	for name, comp := range management.Status.Components {
+		if !comp.Success {
+			failing = append(failing, name)
+		}
+	}
+
+	readyCond := metav1.Condition{
+		Type:               kcm.ReadyCondition,
+		ObservedGeneration: management.Generation,
+	}
+
+	if len(failing) == 0 {
+		readyCond.Status = metav1.ConditionTrue
+		readyCond.Reason = kcm.AllComponentsHealthyReason
+		readyCond.Message = "All components are successfully installed."
+	} else {
+		readyCond.Status = metav1.ConditionFalse
+		readyCond.Reason = kcm.NotAllComponentsHealthyReason
+		readyCond.Message = fmt.Sprintf("Components not ready: %v", failing)
+	}
```

Comment:
```suggestion
	readyCond := metav1.Condition{
		Type:               kcm.ReadyCondition,
		ObservedGeneration: management.Generation,
		Status: metav1.ConditionTrue
		Reason: kcm.AllComponentsHealthyReason
		Message: "All components are successfully installed."
	}

	if len(failing) > 0 {
		readyCond.Status = metav1.ConditionFalse
		readyCond.Reason = kcm.NotAllComponentsHealthyReason
		readyCond.Message = fmt.Sprintf("Components not ready: %v", failing)
	}
```

nit: could be simplified in terms of readability

Reply:
That definitely makes more sense. Thanks.
I would have added your commit but there're few syntax corrections and going for a cleaner git history.